electrode_models/nucleation_single_particle/initialize.py

```diff
@@ -0,0 +1,129 @@
+"""
+    single_particle_electrode.py
+
+    Class file for single-particle (SPM) electrode methods
+"""
+
+import cantera as ct
+import numpy as np
+
+def initialize(input_file, inputs, sep_inputs, counter_inputs, electrode_name, 
+        params, offset):
+    """
+    Initialize the model.
+    """ 
+    class electrode:
+        """
+        Create an electrode object representing the dense electrode
+        """
+
+        # Import relevant Cantera objects.
+        bulk_obj = ct.Solution(input_file, inputs['bulk-phase'])
```

Comment:
The original model you copied this from probably needs better documenting, but just so understand:

- `bulk_obj` = oxide
- `conductor_obj` = carbon

Correct?

Reply:
I made a separate "prod_obj"... I think this needs to be looked at more carefully, maybe next time?